fs/inode/fs_files.c

```diff
@@ -691,17 +766,62 @@ int fs_getfilep(int fd, FAR struct file **filep)
 
   *filep = files_fget(list, fd);
 
-  /* if f_inode is NULL, fd was closed */
+  /* if *filep is NULL, fd was closed */
 
-  if ((*filep)->f_inode == NULL)
+  if (*filep == NULL)
     {
-      *filep = NULL;
       return -EBADF;
     }
 
   return OK;
 }
 
+/****************************************************************************
+ * Name: fs_putfilep
+ *
+ * Description:
+ *   Handles reference counts for files, less than or equal to 0 and close
+ *   the file
+ *
+ * Input Parameters:
+ *   filep  - The caller provided location in which to return the 'struct
+ *            file' instance.
+ ****************************************************************************/
+
+#ifdef CONFIG_FS_REFCOUNT
+int fs_putfilep(FAR struct file *filep)
+{
+  irqstate_t flags;
+  int ret = 0;
+  int refs;
+
+  DEBUGASSERT(filep);
+  flags = spin_lock_irqsave(NULL);
+
+  refs = --filep->f_refs;
+
+  spin_unlock_irqrestore(NULL, flags);
+
+  /* If refs is zero, the close() had called, closing it now. */
+
+  if (refs == 0)
+    {
+      ret = file_close(filep);
+      if (ret < 0)
+        {
+          ferr("ERROR: fs putfilep file_close() failed: %d\n", ret);
+        }
+    }
+
+  return ret;
+}
+#else
+int fs_putfilep(FAR struct file *filep)
```

Comment:
define to null in header file to avoid branch link to this empty function

Reply:
@hujun260 hi, **inline** man, everyone knows that you like to do **inline** implementations, so why don’t you guys review such commits before uploading them from Xiaomi? cc @xiaoxiang781216

Reply:
As above, I can't build locally to verify the compilation error above, I pushed it up to verify the compilation error above. sorry . This code, there is no internal review. I'll fix these problems later @anchao

Reply:
not fix yet.

Reply:
move to fs.h and define to null

Reply:
@Zhangshoukui